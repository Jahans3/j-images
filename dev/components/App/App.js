/**
 * Created by jahansj on 16/11/2016.
 */
import React, { Component } from 'react';
import s from './App.css';

export default class App extends Component {
  constructor() {
    super();

    // Pull out same images each time (see:docs)
    // Use cookies or local storage to remember user's choices
    // get all <content> tags from XML
    // Identify and apply classes

  }

  testApi() {
    const xhr = new XMLHttpRequest();

    xhr.open('GET', 'http://localhost:3030/testApi');

    xhr.onload = () => {
      const res = xhr.response;

      if (!res || xhr.status !== 200) {
        return console.log('Bad response');
      }

      console.log(this.parseXML(res));
    };

    xhr.send();
  }

  parseXML(XML) {
    const parser = new DOMParser();

    return parser.parseFromString(XML, 'application/xml');
  }

  render() {
    return (
        <div>
          It's aliiiiiivee
          <button onClick={this.testApi()}>Test API</button>
        </div>
    )
  }
}