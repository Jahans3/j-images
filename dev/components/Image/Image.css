.listItem {
    display: inline-block;
}

.innerWrap {
    height: 300px;
    width: 300px;
}