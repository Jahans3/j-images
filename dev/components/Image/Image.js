/**
 * Created by jahansj on 19/11/2016.
 */
import React, { Component } from 'react';
import s from './Image.css';

export default class Image extends Component {
  constructor() {
    super();

    this.extractID = this.extractID.bind(this);
    this.toggleSelected = this.toggleSelected.bind(this);

    this.state = {
      selected: {
        isSelected: false,
        className: 'selected',
        id: this.extractID()
      }
    };
  }

  extractID() {
    if (!this.props) return;

    let src = this.props.src;

    src = src.split('/');
    src = src[src.length - 1];

    return src.split('.')[0];
  }

  toggleSelected() {
    const ls = window.localStorage;
    const existing = JSON.parse(localStorage.getItem('flickrImages')) || [];
    const index = existing.indexOf(`imageK${this.props.idKey}`);

    if (index !== -1) {
      existing.splice(index, 1);
      this.setState({ isSelected: false });
    } else {
      existing.push(`imageK${this.props.idKey}`);
      this.setState({ isSelected: true });
    }

    console.log('toggley toggley');
    console.log(ls.getItem('flickrImages'));

    ls.setItem('flickrImages', JSON.stringify(existing));
  }
  
  componentWillMount() {
    // if ID exists in localStorage set state.isSelected to true
    const selectedimages = window.localStorage.getItem('')
  }

  componentDidMount() {
    document.getElementById(`imageK${this.props.idKey}`).addEventListener('click', this.toggleSelected);
  }

  render() {

    return (
        <li className={ s.listItem }>
          <div id={ `imageK${this.props.idKey}` } className={ `${s.innerWrap} ${this.state.isSelected ? 'selected' : ''}` }>
            <h4 className={ s.title }>
              {
                this.props.title
              }
            </h4>
            
            <img src={ this.props.src } alt={ this.props.alt } />
          </div>
        </li>
    )
  }
}