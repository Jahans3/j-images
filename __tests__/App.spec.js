/**
 * Created by jahansj on 19/11/2016.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../dev/components/App/App';
import Image from '../dev/components/Image/Image';

function setState(obj) {
  for (let prop in obj) {
    if (obj.hasOwnProperty(prop)) {
      this[prop] = obj[prop];
    }
  }
}

const addProps = (obj , prop, value) => {
  obj[prop] = value;
};

test('App', () => {
  const tree = renderer.create(<App />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('getData', () => {
  const newApp = {};
  addProps(newApp, 'setState', setState);
  addProps(newApp, 'getData', App.prototype.getData);

  const response = {
    title: 'Title',
    items: [{
      title: 'Image Title',
      media: {
        m: 'http://some.src'
      }
    }]
  };
  const res = JSON.stringify(response);
  const nextImage = (
      <Image
          title={ response.items[0].title }
          src={ response.items[0].media.m }
          alt={ response.items[0].title }
          idKey={0}
          key="0"
      />
  );
  
  newApp.getData(res);
  
  expect(newApp.title).toEqual(response.title);
  expect(newApp.images[0]).toEqual(nextImage);
});