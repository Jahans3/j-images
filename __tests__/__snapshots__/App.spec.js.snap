exports[`test App 1`] = `
<div>
  <div>
    <h1
      className={undefined}>
      
    </h1>
  </div>
  <ul
    className={undefined} />
</div>
`;
